nuxeo-features/nuxeo-platform-oauth/src/main/java/org/nuxeo/ecm/platform/oauth2/clients/OAuth2ClientServiceImpl.java

```diff
@@ -97,4 +137,16 @@ protected DocumentModel getClientModel(String clientId) {
             }
         });
     }
+
+    /**
+     * @since 11.1
+     */
+    protected OAuth2Client execute(Function<Session, OAuth2Client> function) {
+        DirectoryService service = Framework.getService(DirectoryService.class);
+        try (Session session = service.open(OAUTH2CLIENT_DIRECTORY_NAME)) {
```

Comment:
Yes, i agree my first idea was to use execute function by making it generic to avoid many use of Privilege or error handling. But as explained above i was hesitating to do on the same commit, i told my self i wait until PR and if the stuff is validate i will do one the rest of the code as example my current execute method will become 

`protected <T> T execute(Function<Session, T> function) {
        DirectoryService service = Framework.getService(DirectoryService.class);
        return Framework.doPrivileged(() -> {
            try (Session session = service.open(OAUTH2CLIENT_DIRECTORY_NAME)) {
                return function.apply(session);
            } catch (DirectoryException e) {
                throw new NuxeoException("Error while fetching client directory", e);
            }
        });
    }`
and the use case will be 
`protected List<DocumentModel> queryClients() {
        return execute((session) -> session.query(Collections.emptyMap()));
    }`